stl/src/mutex.cpp

```diff
@@ -107,11 +109,12 @@ static _Thrd_result mtx_do_lock(_Mtx_t mtx, const _timespec64* target) { // lock
             }
 
         } else { // check timeout
+            // TRANSITION, ABI: this branch is preserved for `_Mtx_timedlock`

```

Comment:
This is the only branch that has a condition when `current_thread_id = static_cast<long>(GetCurrentThreadId());` becomes an unnecessary overhead. That is, `while(now.tv_sec < target->tv_sec || now.tv_sec == target->tv_sec && now.tv_nsec < target->tv_nsec)` break directly on the first try.
This happens when `"target>0"` (so the second branch is passed), but `"target<=now"` on the first try.

However, I also notice that, as commented in the second branch, we attempt to do a try-lock when the call is `"timed wait for time that has passed"`, but there is no such attempt on this branch. Also, it seems the second branch recognize `"time that has passed"` as `"target<=0"`, but this branch recognize as `"target<=now"`.

I think we could simply ignore this problem and makes no changes here, as this branch is already outdated, and the real fix is to drop this branch entirely when we can safely remove `on the first try`.

Reply:
I believe this should be a do-while loop.

Reply:
Changing control flow is risky. I'm not especially eager to review changes with risk of introducing correctness problems, when they affect separately compiled code that's been working for a long time.

(The background here is that the maintainer team is *severely* overloaded and we don't have a lot of review capacity for the foreseeable future, so I would like to encourage PRs to focus on features and bugfixing, like our libcxx failure backlog, instead of changes to existing code that don't significantly improve observable characteristics.)

Reply:
Resolving conversation without code changes as it is unclear whether anything needs to be added to this PR (which is already getting quite big).